### PR TITLE
Fixes #16710: In HTTPS + syslog mode, agents without the http support (like 5.0) don't have their syslog configured

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/16710

### CODE REVIEW

File: techniques/system/common/1.0/promises.st [hunk 1]
```diff
@@ -278,15 +278,40 @@ bundle agent initialize_ncf()
...
+    (rudder_reporting_syslog|!agent_capability_http).!reports_disabled::
       "configure_syslog"    usebundle => check_log_system;
       "configure_reporting" usebundle => check_rsyslog_version;
 
-    (rudder_reporting_syslog.reports_disabled)|rudder_reporting_https::
+    reports_disabled|(rudder_reporting_https.rsyslog_disabled)::
       "remove_reporting"    usebundle => remove_rudder_syslog_configuration;
 
-    rudder_reporting_https::
+    rudder_reporting_https.agent_capability_http::
```
Comment (Fdall) [inline techniques/system/common/1.0/promises.st]: Isn't it http_reporting ?
See https://github.com/Normation/rudder-packages/blob/master/rudder-agent/SOURCES/Makefile

File: techniques/system/common/1.0/promises.st [hunk 2]
```diff
@@ -278,15 +278,40 @@ bundle agent initialize_ncf()
...
-    rudder_reporting_syslog.!reports_disabled::
+    !rsyslog_disabled.!reports_disabled.policy_server::
+      "define rsyslog configuration for relays" usebundle => rsyslog_system_settings;
+
+    (rudder_reporting_syslog|!agent_capability_http).!reports_disabled::
       "configure_syslog"    usebundle => check_log_system;
       "configure_reporting" usebundle => check_rsyslog_version;
 
-    (rudder_reporting_syslog.reports_disabled)|rudder_reporting_https::
+    reports_disabled|(rudder_reporting_https.rsyslog_disabled)::
```
Comment (Fdall) [inline techniques/system/common/1.0/promises.st]: On a relay will it not remove the syslog report forwarding?

File: techniques/system/common/1.0/promises.st (hunk 1 above)
Comment (ncharles) [reply techniques/system/common/1.0/promises.st]: it is

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
OK, squash merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

File: techniques/system/common/1.0/promises.st (hunk 2 above)
Comment (ncharles) [reply techniques/system/common/1.0/promises.st:288]: it should have been 
reports_disabled|(rudder_reporting_https.agent_capability_http_reporting)|rsyslog_disabled)::

**Pull request merged**